### PR TITLE
rpi-eeprom_git: Update recipe to latest rpi-eeprom repo

### PR DESCRIPTION
**Pull request by mdsketch**
- What I did

I upped the commit version to the latest release of https://github.com/raspberrypi/rpi-eeprom

This will allow updating the pi4 to the latest version eeprom. It does not support the pi5 (but it could with some more editing).

- How I did it

I had to change the directories that are installed as they changed upstream. I matched the compatibility that upstream provides.

Also removed the patch as it's already changed. Here is the upstream change that does the same thing as the patch https://github.com/raspberrypi/rpi-eeprom/blob/5ec5c003bacc73847aadad712aa1fbdace8f1c4e/rpi-eeprom-update#L122

and here is the actual patch commit https://github.com/raspberrypi/rpi-eeprom/commit/55ca589ab9e6e719c1e1f9fc72997d1643d28fe3

PR is the same as #1239 but I followed the correct guidelines

### CODE REVIEW

**Comment by kraj** [comment]
Can you do a rebase on master ( instead of rebase ) so we can avoid unnecessary merge commits ?

**Comment by mdsketch** [comment]
> Can you do a rebase on master ( instead of rebase ) so we can avoid unnecessary merge commits ?

Done, sorry about that.

**Comment by agherzan** [comment]
This looks good. Two small requests to maintain the history assumptions and contribution guidelines:
1. squash the commits while maintaining the git log - in essence they are one logical change
2. I would reword the subject for the resulting commit to maintain the current format: " rpi-eeprom: Update to 2023.10.18-2712" 
Thanks once again for this.

**Comment by agherzan** [comment]
The commit ended up pretty messed up. Here is what I propose to change it to:

```
rpi-eeprom: Update to 2023.10.18-2712

This follows the current latest release of rpi-eeprom:
https://github.com/raspberrypi/rpi-eeprom

This also removes a patch that is already in upstream.
https://github.com/raspberrypi/rpi-eeprom/blob/5ec5c003bacc73847aadad712aa1fbdace8f1c4e/rpi-eeprom-update#L122
https://github.com/raspberrypi/rpi-eeprom/commit/55ca589

Signed-off-by: Matthew Draws <draws.matthew@gmail.com>
```

**Comment by mdsketch** [comment]
Yup makes more sense, I had everything from both commit messages in there

**Review by agherzan — APPROVED** [review]
Thanks for this. Approved and kicked the CI.
